streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamTaskTest.java

```diff
@@ -639,7 +639,31 @@ public void shouldRespectCommitNeeded() {
         assertTrue(task.commitNeeded());
 
         task.commit();
-        assertFalse(task.commitNeeded());
+        assertFalse(task.commitNeeded()); // confirm that timestamp was correctly committed
+        assertTrue(Long.parseLong(task.consumer.committed(partition1).metadata())
```

Comment:
We should not piggy-back test for new features to existing tests. It's better to add a new test method `shouldAddPartitionTimeToOffsetMetatdataOnCommit`

Reply:
Just realized the test I added included the test for commit mechanism anyways.